rust/tvm-macros/src/object.rs

```diff
@@ -185,5 +187,26 @@ pub fn macro_impl(input: proc_macro::TokenStream) -> TokenStream {
 
     expanded.extend(base_tokens);
 
+    if derive {
+        let derives = quote! {
+            impl std::hash::Hash for #ref_id {
+                fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+                    self.0.hash(state)
```

Comment:
Possibly dumb comment (I'm not familiar with this macro) - is `self.0` always the right thing to hash? Or is `#ref_id` sometimes a tuple with multiple fields, where field `.1` and `.2` etc. may be part of what distinguishes objects when hashing?

Reply:
No we generate a wrapper new type over the ObjectPtr<T> which contains 1 field always. ObjectPtr is currently dispatching to TVM's C++ hashing so we have cross language consistency.